### ISSUE TITLE
Relative symlinks to directories broken on windows (fails in GHA windows CI)

### ISSUE THREAD

**Issue by IanButterworth** [JuliaLang/Pkg.jl#2378]
A local tracking issue to explain failing tests on master.

Failing at

https://github.com/JuliaLang/Pkg.jl/blob/f269fd4352a2f45fa5d4d731b2e0f2010e259f7b/test/artifacts.jl#L140

with
```
File permissions: Error During Test at D:\a\Pkg.jl\Pkg.jl\test\artifacts.jl:140
  Test threw exception
  Expression: !(iszero(filemode(joinpath(artifact_dir, dir_link)) & 0x92))
  IOError: stat("C:\\Users\\runneradmin\\AppData\\Local\\Temp\\jl_fElJkZ\\71e3ab93dd8c52c3e630df0d4178aed8f8712598\\dir_link"): permission denied (EACCES)
```

Relative symlinks to directories are broken on windows. 

Upstream fix PR candidate by @staticfloat https://github.com/JuliaLang/julia/pull/39491 which is shown to fix the issue [here](https://github.com/JuliaLang/Pkg.jl/pull/2374/commits/6ae2645b91b4384e249938fd21955aed828833cf)

I'm not sure we know why AV windows is passing. Perhaps only one runs as admin?

**Comment by DilumAluthge** [comment]
I'm now seeing this error on GHA Windows. Is this the same issue, or a different issue?
```
Test Summary:  | Pass  Total
Artifact Usage |    9      9
Test Summary:             | Pass  Total
Artifact GC collect delay |   16     16
Override.toml: Test Failed at C:\hostedtoolcache\windows\julia\nightly\x64\share\julia\stdlib\v1.7\Pkg\test\artifacts.jl:582
  Expression: arty_path == barty_override_path
   Evaluated: "C:\\Users\\runneradmin\\AppData\\Local\\Temp\\jl_Keo6ZZ\\depot2\\artifacts\\64d0b4f8d9c004b862b38c4acfbd74988226995c" == "C:\\Users\\runneradmin\\AppData\\Local\\Temp\\jl_Keo6ZZ\\a_wild_barty_appears"
Stacktrace:
 [1] (::Main.PkgTests.ArtifactTests.var"#117#133"{String})()
   @ Main.PkgTests.ArtifactTests C:\hostedtoolcache\windows\julia\nightly\x64\share\julia\stdlib\v1.7\Pkg\test\artifacts.jl:582
 [2] activate(f::Main.PkgTests.ArtifactTests.var"#117#133"{String}, new_project::String)
   @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Pkg\src\API.jl:1513
 [3] (::Main.PkgTests.ArtifactTests.var"#107#120")(depot_container::String)
   @ Main.PkgTests.ArtifactTests C:\hostedtoolcache\windows\julia\nightly\x64\share\julia\stdlib\v1.7\Pkg\test\artifacts.jl:576
 [4] mktempdir(fn::Main.PkgTests.ArtifactTests.var"#107#120", parent::String; prefix::String)
   @ Base.Filesystem .\file.jl:734
 [5] mktempdir(fn::Function, parent::String)
   @ Base.Filesystem .\file.jl:732
 [6] macro expansion
   @ C:\hostedtoolcache\windows\julia\nightly\x64\share\julia\stdlib\v1.7\Pkg\test\artifacts.jl:457 [inlined]
 [7] macro expansion
   @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.7\Test\src\Test.jl:1152 [inlined]
 [8] top-level scope
   @ C:\hostedtoolcache\windows\julia\nightly\x64\share\julia\stdlib\v1.7\Pkg\test\artifacts.jl:457
Test Summary: | Pass  Fail  Total
Override.toml |   20     1     21
ERROR: LoadError: LoadError: Some tests did not pass: 20 passed, 1 failed, 0 errored, 0 broken.
in expression starting at C:\hostedtoolcache\windows\julia\nightly\x64\share\julia\stdlib\v1.7\Pkg\test\artifacts.jl:1
in expression starting at C:\hostedtoolcache\windows\julia\nightly\x64\share\julia\stdlib\v1.7\Pkg\test\runtests.jl:3
ERROR: Package Pkg errored during testing
```

Log: https://github.com/JuliaLang/Pkg.jl/runs/1867594575

**Comment by IanButterworth** [comment]
That's fixed as of #2390 

We're all green! 🎉
